src/net/net.c

```diff
@@ -196,11 +196,16 @@ static boolean get_static_config(tuple root, struct netif *n, boolean trace) {
     ip4_addr_t netmask;
     ip4_addr_t gw;
 
-    if (!get_config_addr(root, sym(ipaddr), &ip) ||
-        !get_config_addr(root, sym(gateway), &gw) ||
-        !get_config_addr(root, sym(netmask), &netmask))
+    if (!get_config_addr(root, sym(ipaddr), &ip))
         return false;
 
+    if (!get_config_addr(root, sym(gateway), &gw))
```

Comment:
I suggest first checking if the netmask is configured, defaulting to /24 if not, and then afterwards choosing the default gateway (if unconfigured) based on the actual netmask. This will allow the gateway default setting to work if a netmask other than /24 is used.

Also, the stored addresses are in network byte order, so you'll want to make use of lwip_ntohl / lwip_htonl.

Reply:
Fixed. Thanks for the byte order tip, I didn't realize that the web app running in nanos was actually _not_ responding (although the desired IP was assigned, the mask was wrong)